MapboxCoreNavigation/NavigationService.swift

```diff
@@ -268,6 +268,8 @@ public class MapboxNavigationService: NSObject, NavigationService, DefaultInterf
     deinit {
         suspendNotifications()
         endNavigation()
+        nativeLocationSource.delegate = nil
+        simulatedLocationSource?.delegate = nil
```

Comment:
This is an important addition, but I can't help but wonder if there are other places where this needs to be done as well?

Reply:
Apparently `CLLocationManager` is an exception because unlike convention, they store their delegate reference as `unowned` instead of `weak`. Apparently you usually you don't have to worry about this.

I also searched the code -- we don't set location manager delegates outside of the navigation service.